### ISSUE TITLE
bug: not validate csv file

### ISSUE THREAD

**Issue by Pardeep94** [nzambello/react-csv-reader#9]
Csv reader accept every type of file.
but i need to restrict it to read only csv type file

**Comment by furrutiaunicode** [comment]
the csv reader only allow csv type files, how are you using the react-csv-reader?.

this is one example with only csv files allow:

```

import CSVReader from "react-csv-reader";


    <CSVReader
      cssClass="react-csv-input"
      label="Select CSV with secret Death Star statistics"
      onFileLoaded={handleForce}
    />
```

**Comment by Pardeep94** [comment]
I am also doing the same but it accept every type of file.


<CSVReader
              cssClass="react-csv-input"
              onFileLoaded={this.handleImport}
            />

**Comment by furrutiaunicode** [comment]
> I am also doing the same but it accept every type of file.

Can you show your code pleased. you install this for npm?

**Comment by Pardeep94** [comment]
![screenshot from 2018-12-14 15-35-35](https://user-images.githubusercontent.com/18335284/49996886-2a15c280-ffb6-11e8-9804-4f37dd9359a5.jpeg)


This array i received when i upload jpg file.


yes i was install it through npm

**Comment by Pardeep94** [comment]
import CSVReader from 'react-csv-reader';

<CSVReader
              cssClass="react-csv-input"
              onFileLoaded={this.handleImport}
            />

handleImport(data) {
    const headerList = [];
    const colorList = [];

    console.log(data);

    for (const index in data[0]) {
      headerList.push({
        name: data[0][index],
        tag: ''
      });
    }
}


I just need header of csv file thats why i used data[0]

**Comment by furrutiaunicode** [comment]
ok, you can search on node_modules, react-csv-reader -> dist -> react-csv-reader.js and confirm that when you search for accept this only say: 

> accept:"text/csv",

if not, go to https://github.com/nzambello/react-csv-reader/blob/master/dist/react-csv-reader.js and change this code for the code inside the dist.

**Comment by Pardeep94** [comment]
I just check node_modules.
but it is correct.
this check is already exist in node_module

**Comment by furrutiaunicode** [comment]
> I just check node_modules.
> but it is correct.
> this check is already exist in node_module

The problem is that the file declaration to be accepted by the csv reader is currently in that file, try making an update and changing that file for the current one that you have in your project. If that does not work, try to put a normal input file, 

> <input type="file" accept=".csv" />

If even in this input it lets you enter files besides csv, the problem is not the csvreader, and you have a complement or some directive that makes a correction in the input.

**Comment by Pardeep94** [comment]
Now i understand the issue.
only this "accept " validation is not enough because when we are going to upload file then we have a option to change file type in select file popup of computer and after that we can upload any type of file.

To avoid this type of issue we need to manually check the file extension when file is received after upload and if it is not csv than that file will not accepted by CSVReader.

**Comment by furrutiaunicode** [comment]
Excellent. So what you have to do now is to verify or the extension of the file, or depending on what you want to use the file, verify the integrity of it, whether it is the number of columns etc. For example, I verify the number of columns, since mainly I use it to make an import to the database. regards

**Comment by darkwebdev** [comment]
The problem with `input type="file"` is that different browsers show the dialog box differently. Chrome shows only .csv-files and Firefox shows all files unless you use the `accept` attribute like this:
```<input type="file" accept=".csv" />```

**Comment by Pardeep94** [comment]
I am using this extension on Chrome but my operating system is ubuntu and on ubuntu `accept="text/csv"` is not working. only `accept=".csv"` is working in ubuntu chrome

**Comment by nzambello** [comment]
I tested it on both Chrome and Firefox and now it works correctly.
I'm using macOS.

@darkwebdev @Pardeep94 can you test this in development before I manage to release it?

**Comment by darkwebdev** [comment]
Fixed for Firefox for Mac 👍

**Comment by nzambello** [comment]
Thank you all!
Released 1.2.1 version.

**Comment by AmineBlel** [comment]
Hi guys,
still got the same problem, it works correctly on Firefox, but with Chrome no.

I'm using ubuntu 18.04
